app/models/spree/payment_fixes.rb

```diff
@@ -0,0 +1,10 @@
+module Spree
+  module PaymentFixes
+
+    def update_order
+      # without reload order was updated with inaccurate data
+      order.reload
+      super
+    end
+  end
```

Comment:
Let's use the approach mentioned [here](https://gist.github.com/kushniryb/101c161eab8d2d2e2e35e4a17b781d1b) with decorators. 
Also - I think we can rewrite it simply as `order.reload && super`